consensus/istanbul/backend/snapshot.go

```diff
@@ -220,22 +219,13 @@ func (s *Snapshot) apply(headers []*types.Header, gov governance.Engine, addr co
 			}
 		}
 	}
-	snap.Number += uint64(len(headers))
```

Comment:
Because other fields of `snap` directly assign the last header of `headers`, using the block number of the last header is more consistent value